File: backend/src/main/java/com/messmate/backend/dto/request/ExpenseRequest.java
package com.messmate.backend.dto.request;

import com.messmate.backend.entity.ExpenseItem;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.Data;

import java.time.LocalDate;
import java.util.List;

@Data
public class ExpenseRequest {
    @NotBlank
    private String title;

    private String description;

    @NotNull
    private LocalDate date;

    private String category;

    private String mealScope; // LUNCH, DINNER, BOTH

    private String paidBy;

    private String receiptUrl;

    // For manual/custom splits
    private List<CustomSplit> customSplits;

    @NotNull
    private Double totalAmount;

    private String splitMethod = "MEAL_BASED"; // EQUAL, MEAL_BASED

    private List<ExpenseItem> items;
}

File: android/app/src/main/java/com/messmate/android/data/expense/ExpenseRequest.kt
package com.messmate.android.data.expense

data class ExpenseItem(
    val name: String,
    val amount: Double
)

data class CustomSplit(
    val memberId: String,
    val amount: Double? = null,
    val percentage: Double? = null
)

data class ExpenseRequest(
    val title: String,
    val description: String?,
    val date: String,
    val category: String? = null,
    val mealScope: String? = null,
    val paidBy: String? = null,
    val receiptUrl: String? = null,
    val totalAmount: Double,
    val splitMethod: String = "AUTO_MEAL",
    val items: List<ExpenseItem>? = null,
    val customSplits: List<CustomSplit>? = null
)
